### PR TITLE
Add integer exponent code sample to JitBuilder

### PR DESCRIPTION
**Pull request by sudip-unb**
Provide a new JitBuilder sample code that demonstrates how to compute power(a,n) using the DoWhileLoop construct to multiply a by itself n times (n is an integer). This code sample will be included in the set of default code samples that runs when OMR_JITBUILDERTEST is enabled in cmake builds.

### CODE REVIEW

File: jitbuilder/release/cpp/samples/Power.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+      {
+      fprintf(stderr, "FAIL: could not initialize JIT\n");
+      exit(-1);
+      }
+
+   printf("Step 2: define relevant types\n");
+   OMR::JitBuilder::TypeDictionary types;
+
+   printf("Step 3: compile method builder\n");
+   PowerMethod PowerMethod(&types);
```
Comment (mstoodle) [inline jitbuilder/release/cpp/samples/Power.cpp:93]: shouldn't the variable have a different name than the class? i.e. `PowerMethod powerMethod(&types);`

**Review by mstoodle — CHANGES_REQUESTED** [review]
Thank you for your contribution, Sudip, it's great to see more code samples being added for JitBuilder!

Please put your commit title into the active voice, e.g. "Add JitBuilder power function sample:"

There are two sign-offs in your commit (you only need one), and please fix the typo in your commit summary: "jirbuilder".

I think your commit summary could include a bit more information about how the sample works, i.e. "calculates integer powers by iterating over the exponent" or something like that.

File: jitbuilder/release/cpp/samples/Power.cpp (hunk 1 above)
Comment (sudip-unb) [reply jitbuilder/release/cpp/samples/Power.cpp]: Done

**Comment by mstoodle** [comment]
Hi @sudip-unb, the code looks good now. Would you please add a bit more information to the commit message on what the code does and (high level) how it computes the result as I suggested earlier?

**Comment by sudip-unb** [comment]
Hi @mstoodle,
I have updated the commit message at a high level. Would you please have a look? 
Thanks in advance.

**Comment by mstoodle** [comment]
Hi @sudip-unb I see the updated description in the PR, but the commit message still looks like:
```
Add JitBuilder power function sample

Signed-off-by: sudip-unb <sudip.chatterjee@unb.ca>
```

i.e. it has only a subject and is otherwise empty...

**Comment by mstoodle** [comment]
Just so it doesn't look like this one has been left idling (should have documented this back in mid September): I reached out to @sudip-unb directly to explain what it is I'm looking for in the commit message. Once that change is made, I think this PR is good to go.

**Comment by mstoodle** [comment]
The changes look good to me now. I've asked @sudip-unb to rebase onto the latest, at which point I'll run more tests and merge it (assuming they pass :) ).

**Comment by mstoodle** [comment]
jenkins build all

**Comment by sudip-unb** [comment]
@mstoodle rebase is done and the build is successful.

**Comment by mstoodle** [comment]
jenkins build all

**Comment by mstoodle** [comment]
The failure looks like #6571 which has been failing for a while now ( :( ). I'm approving and will merge.

Thanks @sudip-unb !

**Review by mstoodle — APPROVED** [review]

**Pull request merged**